### PR TITLE
Added disengage alert in modeld

### PR DESCRIPTION
**Pull request by mitchellgoffpc**
Added logic to check for spikes in disengagement probability and trigger a corresponding alert.

### CODE REVIEW

**Comment by HaraldSchafer** [comment]
Looks much more readable without the ring buffer!

File: selfdrive/modeld/models/driving.cc [hunk 1]
```diff
@@ -46,8 +46,29 @@ constexpr int OUTPUT_SIZE =  POSE_IDX + POSE_SIZE;
...
+uint64_t last_steering_frame = 0;
+uint64_t history_size = 0;
+
 // #define DUMP_YUV
 
+float smooth(float p, float a, float b) {
+  return p * a + (1 - p) * b;
+}
+
+float threshold_mse(float x, float threshold, bool gt) {
```
Comment (HaraldSchafer) [inline selfdrive/modeld/models/driving.cc:67]: What does gt mean?

Comment (mitchellgoffpc) [reply selfdrive/modeld/models/driving.cc:67]: greater than / less than, it's a flag to control whether the error should be taken above or below the threshold. I can split this up into two separate functions to make it more readable, or maybe just inline it.

Comment (HaraldSchafer) [reply selfdrive/modeld/models/driving.cc:67]: Yeah two functions sounds more readable

**Comment by mitchellgoffpc** [comment]
Closing this PR since the new model outputs give a much easier way to detect collision risk

**Pull request closed without merging**